### PR TITLE
#2515 Added ambiguous constructors to ambiguous constructor error mes…

### PR DESCRIPTION
**Pull request by basclaessen**
Added ambiguous constructors to ambiguous constructor error message

Fixes: #2515

### CODE REVIEW

**Comment by filiphr** [comment]
Good catch @basclaessen. Looking at the tests, I am not sure how I feel about eclipse 😄 . I would rather craft the elements on our side appropriately, so our Eclipse users also see better messages 😄.

Can you please do that as well?

**Comment by basclaessen** [comment]
@filiphr Sure!
I changed the constructors to string conversion. The error message when using the Eclipse compiler is now equal to the error message when using the JDK compiler.

File: processor/src/main/java/org/mapstruct/ap/internal/model/BeanMappingMethod.java [hunk 1]
```diff
@@ -699,7 +700,14 @@ private ConstructorAccessor getConstructorAccessor(Type type) {
                     method.getExecutable(),
                     GENERAL_AMBIGUOUS_CONSTRUCTORS,
                     type,
-                    Strings.join( constructors, ", " )
+                    constructors.stream()
+                        .map( ExecutableElement::getParameters )
+                        .map( ps -> ps.stream()
+                            .map( VariableElement::asType )
+                            .map( String::valueOf )
+                            .collect( Collectors.joining( ",", type.getName() + "(", ")" ) )
```
Comment (filiphr) [inline processor/src/main/java/org/mapstruct/ap/internal/model/BeanMappingMethod.java:708]:
```suggestion
                            .collect( Collectors.joining( ", ", type.getName() + "(", ")" ) )
```

File: processor/src/test/java/org/mapstruct/ap/test/constructor/erroneous/ErroneousConstructorTest.java [hunk 2]
```diff
@@ -29,8 +29,10 @@
             kind = javax.tools.Diagnostic.Kind.ERROR,
             line = 17,
             message = "Ambiguous constructors found for creating org.mapstruct.ap.test.constructor.erroneous" +
-                ".ErroneousAmbiguousConstructorsMapper.PersonWithMultipleConstructors. Either declare parameterless " +
-                "constructor or annotate the default constructor with an annotation named @Default."
+                ".ErroneousAmbiguousConstructorsMapper.PersonWithMultipleConstructors: " +
+                "PersonWithMultipleConstructors(java.lang.String), " +
+                "PersonWithMultipleConstructors(java.lang.String,int). Either declare parameterless constructor " +
```
Comment (filiphr) [inline processor/src/test/java/org/mapstruct/ap/test/constructor/erroneous/ErroneousConstructorTest.java:34]:
```suggestion
                "PersonWithMultipleConstructors(java.lang.String, int). Either declare parameterless constructor " +
```

**Review by filiphr — COMMENTED** [review]
Thanks for the changes @basclaessen. I've added some small suggestion (which I will accept and then merge this PR).

**Pull request merged**